db/wal_dump.cc

```diff
@@ -0,0 +1,39 @@
+#include "rocksdb/wal_dump.h"
+#include <memory>
+#include <thread>
+
+#include "rocksdb/env.h"
+#include "db/wal_manager.h"
+#include "db/version_set.h"
+#include "options/db_options.h"
+
+namespace rocksdb {
+
+#ifndef ROCKSDB_LITE
+
+Status WalDump::CreateWalIter(std::string dbname,
+                                   SequenceNumber start_seq,
+                                   SequenceNumber end_seq,
+                                   std::unique_ptr<TransactionLogIterator>* iter,
+                                   const TransactionLogIterator::ReadOptions& read_options) {
+  thread_local ImmutableDBOptions db_options;
+  thread_local EnvOptions env_options;
+
+  db_options.db_paths.emplace_back(dbname, std::numeric_limits<uint64_t>::max());
+  db_options.wal_dir = dbname;
+  db_options.env = Env::Default();
```

Comment:
This means this can work only on local file system, which is an unnecessary limitation.

Reply:
It is not a limitation, it is used to assign 'env'. I have no idea about other assignments other than Env::Default().